packages/hop-node/src/gasboost/GasBoostTransaction.ts

```diff
@@ -0,0 +1,297 @@
+    }
+    this.chainSlug = chainSlug
+    return this.chainSlug
+  }
+
+  @queue
+  @rateLimitRetry
+  async send () {
+    const nonce = await this.signer.getTransactionCount('pending')
+    this.nonce = nonce
+    const tx = await this.signer.sendTransaction({
+      from: this.from,
+      to: this.to,
+      data: this.data,
+      value: this.value,
+      nonce: this.nonce,
+      gasPrice: this.gasPrice,
+      gasLimit: this.gasLimit
+    })
+
+    this.from = tx.from
+    this.to = tx.to
+    this.data = tx.data
+    this.value = tx.value
+    this.gasLimit = tx.gasLimit
+    this.gasPrice = tx.gasPrice
+
+    this.track(tx)
+    return this
+  }
+
+  async getBumpedGasPrice (): Promise<BigNumber> {
+    const marketGasPrice = await this.signer.getGasPrice()
+    const multiplier = this.gasPriceMultiplier
+    const bumpedGasPrice = getBumpedGasPrice(this.gasPrice, multiplier)
+    return marketGasPrice.lt(bumpedGasPrice) ? bumpedGasPrice : marketGasPrice
+  }
+
+  private handleConfirmation (txHash: string) {
+    this.txHash = txHash
+    this.clearInflightTxs()
+    const receipt = this.getReceipt(txHash)
+    this.emit(State.Confirmed, receipt)
+  }
+
+  private async getReceipt (txHash: string) {
+    return this.signer.provider.waitForTransaction(txHash)
+  }
+
+  async wait (): Promise<providers.TransactionReceipt> {
+    if (this.txHash) {
+      return this.getReceipt(this.txHash)
+    }
+    for (const { hash } of this.inflightItems) {
+      this.getReceipt(hash)
+        .then(() => this.handleConfirmation(hash))
+    }
+    return new Promise((resolve) => {
+      this.on(State.Confirmed, (tx) => {
+        resolve(tx)
+      })
+    })
+  }
+
+  getInflightItems (): InflightItem[] {
+    return this.inflightItems
+  }
+
+  private async startPoller () {
+    if (!this.started) {
+      return
+    }
+    this.started = true
+    while (true) {
+      await this.poll()
+      await wait(this.pollMs)
+    }
+  }
+
+  private async poll () {
+    for (const item of this.inflightItems) {
+      await this.handleInflightTx(item)
+    }
+  }
+
+  private async handleInflightTx (item: InflightItem) {
+    if (item.confirmed) {
+      this.handleConfirmation(item.hash)
+      return
+    }
+    if (!this.shouldBoost(item)) {
+      return
+    }
+    await this.boost(item)
+  }
+
+  private shouldBoost (item: InflightItem) {
+    return item.sentAt < (Date.now() - this.timeTilBoostMs)
+  }
+
+  private async boost (item: InflightItem) {
+    const gasPrice = await this.getBumpedGasPrice()
+    if (gasPrice.gt(this.maxGasPriceGwei)) {
+      return
+    }
+    const tx = await this.signer.sendTransaction({
+      from: this.from,
+      to: this.to,
+      data: this.data,
+      value: this.value,
+      nonce: this.nonce,
+      gasPrice,
+      gasLimit: this.gasLimit
+    })
+    this.gasPrice = tx.gasPrice
+    this.boostIndex++
+    this.emit(State.Boosted, tx)
```

Comment:
It does not appear as though anything handles this emission

Reply:
this will be for outside subscribers, like for testing